### ISSUE TITLE
QueryDSL-SQL converts column named "Class" to "class"

### ISSUE THREAD

**Issue by bennesher** [querydsl/querydsl#1980]
I'm using PostgreSQL database, where the case of the identifier matters, and the PostgreSQLTemplates with identifier quoting on.  

For legacy reasons, my schema uses leading caps.  The generated Q-class correctly initializes the column metadata with column name "Class", but the generated query references it as "class".  All other column names come through with their capitalization intact, so this must be getting addressed as a special case somewhere (maybe to reference .class in JPA-QL?)

**Comment by timowest** [comment]
Which Querydsl version do you use?

**Comment by bennesher** [comment]
I'm using 4.1.3.

**Comment by timowest** [comment]
Could you post the Q-class?

**Comment by bennesher** [comment]
I had worked around this by renaming the underlying database column from "Class" to "GradClass".  When reverting the code to the way it was when I encountered the problem, I discovered that the problem was on my side after all: I was using `Expressions.path(Class<?>,Path<?>,String)`  to compose the column reference, and used `"class"` (which was consistent with the naming of other columns), rather than `"classCol"` as the column name.

So the only possible issue might be that QueryDsl silently allowed this reference to an unknown column, rather than issuing a warning or throwing an exception.  But that's a separate point, and I suppose subject to design philosophy.
